report/src/main/webapp/app/components/compareScreens.directive.js

```diff
@@ -0,0 +1,648 @@
+        wrapper.innerHTML = '</div><div class="difs"></div><div class="customMasks"><div class="customMask maskAtop"></div><div class="customMask maskAbot"></div><div class="customMask maskBtop"></div><div class="customMask maskBbot"></div><div class="customLabel labelA"><div class="ruler"></div><div class="text"></div></div><div class="customLabel labelB"><div class="ruler"></div><div class="text"></div></div></div><div class="canvas-wrapper"></div>';
+      }
+
+      if (!wrapper) {
+        console.warn('Can\'t find wrapper');
+        return;
+      } else if (wrapper.classList.contains(initedClass)) {
+        console.warn('Screen comparison already has already been initialized');
+        return;
+      } else {
+        wrapper.classList.add(initedClass);
+      }
+
+      var button = document.querySelector('.try-new');
+      button.classList.add('button-disabled');
+
+      var infoMsg = document.createElement('div');
+      infoMsg.innerHTML = '<div class="info-msg">Advanced screen comparison in progress - please wait</div>';
+      var tabContent = document.querySelector('.tab-content');
+      tabContent.appendChild(infoMsg);
+
+      var items = wrapper.querySelectorAll('.layout-compare-item .img-responsive:not(.mask)');
+
+      if (items.length !== 2) {
+        return;
+      }
+
+      prepareMarkup(wrapper);
+
+      var img = {
+        imgA: items[0].src,
+        imgB: items[1].src,
+      };
+
+      var canvas = {
+        canvasWrapper: wrapper.querySelector('.canvas-wrapper'),
+        canvasA: document.createElement('canvas'),
+        canvasB: document.createElement('canvas'),
+      };
+
+      var label = {
+        labelA: wrapper.querySelector('.labelA'),
+        labelB: wrapper.querySelector('.labelB'),
+      };
+
+      var ruler = {
+        rulerA: label.labelA.querySelector('.ruler'),
+        rulerB: label.labelB.querySelector('.ruler'),
+      };
+
+      var mask = {
+        maskA: {
+          top: wrapper.querySelector('.maskAtop'),
+          bot: wrapper.querySelector('.maskAbot'),
+        },
+        maskB: {
+          top: wrapper.querySelector('.maskBtop'),
+          bot: wrapper.querySelector('.maskBbot'),
+        },
+        customMask: wrapper.querySelector('.customMasks'),
+        maskSize: null,
+      };
+
+      var context = {
+        contextA: canvas.canvasA.getContext('2d'),
+        contextB: canvas.canvasB.getContext('2d'),
+      };
+
+      var imgSize = {
+        imgSizeA: null,
+        imgSizeB: null,
+      };
+
+      var simple = {
+        simpleA: null,
+        simpleB: null,
+      };
+
+      var cursor = {
+        cursorA: 0,
+        cursorB: 0,
+        cursorBstart: 0,
+      };
+
+      var difs = {
+        main: wrapper.querySelector('.difs'),
+        difsA: [],
+        difsB: [],
+      };
+
+      var invertedGroups = {
+        invertedGroupsA: [],
+        invertedGroupsB: [],
+      };
+
+      var groups = [];
```

Comment:
Could you please make those variables more meaningful?

Reply:
Done